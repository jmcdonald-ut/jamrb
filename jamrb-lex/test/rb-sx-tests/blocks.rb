10.times do
  puts "Hello"
end

hash = {
  who: "I.B. Banker",
  where: "Bank",
  what: "phishing with friends"
}

hash.each do |key, val|
  3.times do
    puts "#{key} => #{val}"
  end
end

local = "hello"
hash.each do |key, value, ; local|
  local = "good bye"
  puts "#{key} => #{value} - #{local}"
  local = "hi"
end
