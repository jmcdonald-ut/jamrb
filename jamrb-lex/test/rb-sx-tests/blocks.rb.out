((Token ((1 0) on_int "10"))
 (Token ((1 2) on_period "."))
 (Token ((1 3) on_ident "times"))
 (Token ((1 8) on_sp " "))
 (Token ((1 9) on_kw "do"))
 (Token ((1 11) on_ignored_nl "\\n"))
 (Token ((2 0) on_sp "  "))
 (Token ((2 2) on_ident "puts"))
 (Token ((2 6) on_sp " "))
 (Token ((2 7) on_tstring_beg "\\\""))
 (Token ((2 8) on_tstring_content "Hello"))
 (Token ((2 13) on_tstring_end "\\\""))
 (Token ((2 14) on_nl "\\n"))
 (Token ((3 0) on_kw "end"))
 (Token ((3 3) on_nl "\\n"))
 (Token ((4 0) on_ignored_nl "\\n"))
 (Token ((5 0) on_ident "hash"))
 (Token ((5 4) on_sp " "))
 (Token ((5 5) on_op "="))
 (Token ((5 6) on_sp " "))
 (Token ((5 7) on_lbrace "{"))
 (Token ((5 8) on_ignored_nl "\\n"))
 (Token ((6 0) on_sp "  "))
 (Token ((6 2) on_label "who:"))
 (Token ((6 6) on_sp " "))
 (Token ((6 7) on_tstring_beg "\\\""))
 (Token ((6 8) on_tstring_content "I.B. Banker"))
 (Token ((6 19) on_tstring_end "\\\""))
 (Token ((6 20) on_comma ","))
 (Token ((6 21) on_ignored_nl "\\n"))
 (Token ((7 0) on_sp "  "))
 (Token ((7 2) on_label "where:"))
 (Token ((7 8) on_sp " "))
 (Token ((7 9) on_tstring_beg "\\\""))
 (Token ((7 10) on_tstring_content "Bank"))
 (Token ((7 14) on_tstring_end "\\\""))
 (Token ((7 15) on_comma ","))
 (Token ((7 16) on_ignored_nl "\\n"))
 (Token ((8 0) on_sp "  "))
 (Token ((8 2) on_label "what:"))
 (Token ((8 7) on_sp " "))
 (Token ((8 8) on_tstring_beg "\\\""))
 (Token ((8 9) on_tstring_content "phishing with friends"))
 (Token ((8 30) on_tstring_end "\\\""))
 (Token ((8 31) on_nl "\\n"))
 (Token ((9 0) on_rbrace "}"))
 (Token ((9 1) on_nl "\\n"))
 (Token ((10 0) on_ignored_nl "\\n"))
 (Token ((11 0) on_ident "hash"))
 (Token ((11 4) on_period "."))
 (Token ((11 5) on_ident "each"))
 (Token ((11 9) on_sp " "))
 (Token ((11 10) on_kw "do"))
 (Token ((11 12) on_sp " "))
 (Token ((11 13) on_op "|"))
 (Token ((11 14) on_ident "key"))
 (Token ((11 17) on_comma ","))
 (Token ((11 18) on_sp " "))
 (Token ((11 19) on_ident "val"))
 (Token ((11 22) on_op "|"))
 (Token ((11 23) on_ignored_nl "\\n"))
 (Token ((12 0) on_sp "  "))
 (Token ((12 2) on_int "3"))
 (Token ((12 3) on_period "."))
 (Token ((12 4) on_ident "times"))
 (Token ((12 9) on_sp " "))
 (Token ((12 10) on_kw "do"))
 (Token ((12 12) on_ignored_nl "\\n"))
 (Token ((13 0) on_sp "    "))
 (Token ((13 4) on_ident "puts"))
 (Token ((13 8) on_sp " "))
 (Token ((13 9) on_tstring_beg "\\\""))
 (Token ((13 10) on_embexpr_beg "\\#{"))
 (Token ((13 12) on_ident "key"))
 (Token ((13 15) on_embexpr_end "}"))
 (Token ((13 16) on_tstring_content " => "))
 (Token ((13 20) on_embexpr_beg "\\#{"))
 (Token ((13 22) on_ident "val"))
 (Token ((13 25) on_embexpr_end "}"))
 (Token ((13 26) on_tstring_end "\\\""))
 (Token ((13 27) on_nl "\\n"))
 (Token ((14 0) on_sp "  "))
 (Token ((14 2) on_kw "end"))
 (Token ((14 5) on_nl "\\n"))
 (Token ((15 0) on_kw "end"))
 (Token ((15 3) on_nl "\\n"))
 (Token ((16 0) on_ignored_nl "\\n"))
 (Token ((17 0) on_ident "local"))
 (Token ((17 5) on_sp " "))
 (Token ((17 6) on_op "="))
 (Token ((17 7) on_sp " "))
 (Token ((17 8) on_tstring_beg "\\\""))
 (Token ((17 9) on_tstring_content "hello"))
 (Token ((17 14) on_tstring_end "\\\""))
 (Token ((17 15) on_nl "\\n"))
 (Token ((18 0) on_ident "hash"))
 (Token ((18 4) on_period "."))
 (Token ((18 5) on_ident "each"))
 (Token ((18 9) on_sp " "))
 (Token ((18 10) on_kw "do"))
 (Token ((18 12) on_sp " "))
 (Token ((18 13) on_op "|"))
 (Token ((18 14) on_ident "key"))
 (Token ((18 17) on_comma ","))
 (Token ((18 18) on_sp " "))
 (Token ((18 19) on_ident "value"))
 (Token ((18 24) on_comma ","))
 (Token ((18 25) on_sp " "))
 (Token ((18 26) on_semicolon ";"))
 (Token ((18 27) on_sp " "))
 (Token ((18 28) on_ident "local"))
 (Token ((18 33) on_op "|"))
 (Token ((18 34) on_ignored_nl "\\n"))
 (Token ((19 0) on_sp "  "))
 (Token ((19 2) on_ident "local"))
 (Token ((19 7) on_sp " "))
 (Token ((19 8) on_op "="))
 (Token ((19 9) on_sp " "))
 (Token ((19 10) on_tstring_beg "\\\""))
 (Token ((19 11) on_tstring_content "good bye"))
 (Token ((19 19) on_tstring_end "\\\""))
 (Token ((19 20) on_nl "\\n"))
 (Token ((20 0) on_sp "  "))
 (Token ((20 2) on_ident "puts"))
 (Token ((20 6) on_sp " "))
 (Token ((20 7) on_tstring_beg "\\\""))
 (Token ((20 8) on_embexpr_beg "\\#{"))
 (Token ((20 10) on_ident "key"))
 (Token ((20 13) on_embexpr_end "}"))
 (Token ((20 14) on_tstring_content " => "))
 (Token ((20 18) on_embexpr_beg "\\#{"))
 (Token ((20 20) on_ident "value"))
 (Token ((20 25) on_embexpr_end "}"))
 (Token ((20 26) on_tstring_content " - "))
 (Token ((20 29) on_embexpr_beg "\\#{"))
 (Token ((20 31) on_ident "local"))
 (Token ((20 36) on_embexpr_end "}"))
 (Token ((20 37) on_tstring_end "\\\""))
 (Token ((20 38) on_nl "\\n"))
 (Token ((21 0) on_sp "  "))
 (Token ((21 2) on_ident "local"))
 (Token ((21 7) on_sp " "))
 (Token ((21 8) on_op "="))
 (Token ((21 9) on_sp " "))
 (Token ((21 10) on_tstring_beg "\\\""))
 (Token ((21 11) on_tstring_content "hi"))
 (Token ((21 13) on_tstring_end "\\\""))
 (Token ((21 14) on_nl "\\n"))
 (Token ((22 0) on_kw "end"))
 (Token ((22 3) on_nl "\\n")))
